core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -250,10 +266,114 @@ public void setConf(Configuration conf) {
     this.conf = conf;
   }
 
+  private boolean insertProperties(Namespace namespace, Map<String, String> properties) {
+    String namespaceName = JdbcUtil.namespaceToString(namespace);
+
+    try {
+      int insertedRecords = connections.run(conn -> {
+        StringBuilder sqlStatement = new StringBuilder(JdbcUtil.INSERT_NAMESPACE_PROPERTIES_SQL);
+
+        for (int i = 0; i < properties.size(); i++) {
+          if (i != 0) {
+            sqlStatement.append(", ");
+          }
+          sqlStatement.append(JdbcUtil.INSERT_PROPERTIES_VALUES_BASE);
+        }
+
+        try (PreparedStatement sql = conn.prepareStatement(sqlStatement.toString())) {
+          int rowIndex = 0;
+          for (Map.Entry<String, String> keyValue : properties.entrySet()) {
+            sql.setString(rowIndex + 1, catalogName);
+            sql.setString(rowIndex + 2, namespaceName);
+            sql.setString(rowIndex + 3, keyValue.getKey());
+            sql.setString(rowIndex + 4, keyValue.getValue());
+            rowIndex += 4;
+          }
+          return sql.executeUpdate();
+        }
+      });
+
+      if (insertedRecords == properties.size()) {
+        LOG.debug("Successfully committed to new namespace: {}", namespaceName);
+        return true;
+      } else {
+        throw new CommitFailedException("Failed to insertProperties to namespace %s in catalog %s", namespaceName,
+                catalogName);
+      }
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new UncheckedInterruptedException(e,
+              "Interrupted in call to insertProperties(namespace, properties) Namespace: %s", namespace);
```

Comment:
nit: prefer to not use code but human friendly language in error message, like "Interrupted when inserting properties to namespace: %s"

Reply:
please also check your other error messages if they have the same issue, thanks!

Reply:
Agreed but the other methods have the same style as well 
```
Interrupted in call to namespaceExists(namespace)
Interrupted in call to listNamespaces(namespace) Namespace
```
Do we want to change them all to be consistent? I followed what was in the file already. Let me know.

Reply:
Don't worry about the interrupted exceptions. I would probably remove those in general because we will get a stack trace anyway. But it isn't worth making this likely to conflict with other changes.